OemPkg/Library/DeviceBootManagerLib/DeviceBootManagerLib.c

```diff
@@ -0,0 +1,959 @@
+/** @file
+ *DeviceBootManager  - Ms Device specific extensions to BdsDxe.
+
+Copyright (C) Microsoft Corporation. All rights reserved.
+SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Uefi.h>
+
+#include <DfciSystemSettingTypes.h>
+
+#include <Guid/EventGroup.h>
+#include <Guid/GlobalVariable.h>
+#include <Guid/MsBootMenuGuid.h>
+#include <Guid/DfciPacketHeader.h>
+
+#include <Protocol/OnScreenKeyboard.h>
+#include <Protocol/TpmPpProtocol.h>
+
+#include <Library/BaseMemoryLib.h>
+#include <Library/ConsoleMsgLib.h>
+#include <Library/DebugLib.h>
+#include <Library/DeviceBootManagerLib.h>
+#include <Library/DevicePathLib.h>
+#include <Library/HobLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/MsBootManagerSettingsLib.h>
+#include <Library/MsBootOptionsLib.h>
+#include <Library/MsBootPolicyLib.h>
+#include <Library/BootGraphicsLib.h>
+#include <Library/GraphicsConsoleHelperLib.h>
+#include <Library/MsPlatformDevicesLib.h>
+#include <Library/MsPlatformPowerCheckLib.h>
+#include <Library/MsNetworkDependencyLib.h>
+#include <Library/PcdLib.h>
+#include <Library/PrintLib.h>
+#include <Library/PowerServicesLib.h>
+#include <Library/ThermalServicesLib.h>
+#include <Library/UefiBootServicesTableLib.h>
+#include <Library/UefiLib.h>
+#include <Library/UefiRuntimeServicesTableLib.h>
+#include <Library/MsNVBootReasonLib.h>
+#include <Library/MuTelemetryHelperLib.h>
+#include <Library/VariablePolicyHelperLib.h>
+#include <Library/UpdateFacsHardwareSignatureLib.h>
+
+#include <Settings/BootMenuSettings.h>
+#include <Settings/DfciSettings.h>
+
+static EFI_EVENT  mPreReadyToBootEvent;
+static EFI_EVENT  mPostReadyToBootEvent;
+
+CHAR8  mMemoryType[][30] = {
+  // Value for PcdMemoryMapTypes
+  "EfiReservedMemoryType      ",                               // 0x0001
+  "EfiLoaderCode              ",                               // 0x0002
+  "EfiLoaderData              ",                               // 0x0004
+  "EfiBootServicesCode        ",                               // 0x0008
+  "EfiBootServicesData        ",                               // 0x0010
+  "EfiRuntimeServicesCode     ",                               // 0x0020
+  "EfiRuntimeServicesData     ",                               // 0x0040
+  "EfiConventionalMemory      ",                               // 0x0080
+  "EfiUnusableMemory          ",                               // 0x0100
+  "EfiACPIReclaimMemory       ",                               // 0x0200   Both ACPI types would
+  "EfiACPIMemoryNVS           ",                               // 0x0400   be 0x0600
+  "EfiMemoryMappedIO          ",                               // 0x0800
+  "EfiMemoryMappedIOPortSpace ",                               // 0x1000
+  "EfiPalCode                 ",                               // 0x2000
+  "EfiMaxMemoryType           "
+};
+
+static

```

Comment:
nit: should we correct these non UEFI style lines?

Reply:
This was a move with no change.